gwpy/timeseries/timeseries.py

```diff
@@ -1601,6 +1601,87 @@ def convolve(self, fir, window='hanning'):
+                  overwhiten=False, wduration=2, highpass=None, **psd_kw):
+        """Cross-correlate this `TimeSeries` with another signal
+
+        Parameters
+        ----------
+        mfilter : `~numpy.ndarray` or `TimeSeries`
+            the time domain signal to correlate with
+
+        window : `str`, optional
+            window function to apply to timeseries prior to FFT,
+            default: ``'hanning'``
+            see :func:`scipy.signal.get_window` for details on acceptable
+            formats
+
+        detrend : `str`, optional
+            type of detrending to do before FFT (see `~TimeSeries.detrend`
+            for more details), default: ``'constant'``
+
+        overwhiten : `bool`, optional
+            boolean switch to enable (`True`) or disable (`False`) data
+            overwhitening, default: `False`
+
+        wduration : `float`, optional
+            duration (in seconds) of the time-domain FIR whitening filter,
+            only used if `overwhiten=True`, defaults to 2 seconds
+
+        highpass : `float`, optional
+            highpass corner frequency (in Hz) of the FIR whitening filter,
+            only used if `overwhiten=True`, default: `None`
+
+        **psd_kw
+            keyword arguments to pass to `TimeSeries.psd` to generate
+            a PSD, only used if `overwhiten=True`
+
+        Returns
+        -------
+        snr : `TimeSeries`
+            the correlated signal-to-noise ratio (SNR) timeseries
+
+        See Also
+        --------
+        TimeSeries.psd
+            for details on the PSD calculation
+        TimeSeries.convolve
+            for details on convolution with the overlap-save method
+
+        Notes
+        -----
+        The `window` argument is used in PSD estimation, overwhitening, and
+        preventing spectral leakage in the output.
+
+        Due to filter settle-in, a segment half the length of `mfilter` will be
+        corrupted at the beginning and end of the output. See
+        `~TimeSeries.convolve` for more details.
+
+        The output SNR is positive-definite, detrended, and normalised by its
+        standard deviation, so that the SNR measures number of standard
+        deviations above the mean.
+        """
+        # condition data
+        if overwhiten:
+            fftlength = psd_kw.pop('fftlength',
+                                   _fft_length_default(self.dt))
+            overlap = psd_kw.pop('overlap', None)
+            if overlap is None:
+                overlap = recommended_overlap(window) * fftlength
+            psd = self.psd(fftlength, overlap, window=window, **psd_kw)
+            # overwhiten (with errors on division by zero)
+            with numpy.errstate(all='raise'):
+                in_ = self.copy().whiten(window=window, asd=psd,
+                                         fduration=wduration,
+                                         highpass=highpass)
+        else:
+            in_ = self.copy()
```

Comment:
why is this `copy()` required?

Reply:
I think I was attempting to be safe by not performing any operations in-place. Would it be better to replace `in_ = self.copy().whiten(...)` with `self.whiten(...)` on line 1674, and to replace `in_ = self.copy()` with `in_ = self` on line 1678?

Reply:
This is done in the most recent commit, let me know if you like this change better.